iceoryx2-cal/src/resizable_shared_memory/dynamic.rs

```diff
@@ -0,0 +1,698 @@
+            if let Some((extracted_name, _)) = Self::extract_name_and_segment_id(raw_name) {
+                if *name == extracted_name {
+                    fail!(from origin, when Shm::remove_cfg(raw_name, config),
+                        "{msg} since the underlying SharedMemory could not be removed.");
+                    shm_removed = true;
+                }
+            }
+        }
+
+        Ok(shm_removed)
+    }
+
+    fn does_exist_cfg(
+        name: &FileName,
+        config: &Shm::Configuration,
+    ) -> Result<bool, NamedConceptDoesExistError> {
+        let origin = "resizable_shared_memory::Dynamic::does_exist_cfg()";
+        let msg = format!(
+            "Unable to determine if ResizableSharedMemory {:?} exists",
+            name
+        );
+
+        let mgmt_name = Self::managment_segment_name(name);
+        Ok(
+            fail!(from origin, when Shm::does_exist_cfg(&mgmt_name, config),
+            "{msg} since the existance of the underlying managment segment could not be verified."),
+        )
+    }
+
+    fn list_cfg(config: &Shm::Configuration) -> Result<Vec<FileName>, NamedConceptListError> {
+        let origin = "resizable_shared_memory::Dynamic::list_cfg()";
+        let mut names = vec![];
+        let raw_names = fail!(from origin, when Shm::list_cfg(config),
+                            "Unable to list ResizableSharedMemories since the underlying SharedMemories could not be listed.");
+
+        for raw_name in &raw_names {
+            if let Some(name) = Self::extract_name_from_management_segment(raw_name) {
+                names.push(name);
+            }
+        }
+
+        Ok(names)
+    }
+
+    fn remove_path_hint(value: &Path) -> Result<(), super::NamedConceptPathHintRemoveError> {
+        Shm::remove_path_hint(value)
+    }
+}
+
+impl<Allocator: ShmAllocator, Shm: SharedMemory<Allocator>> DynamicMemory<Allocator, Shm>
+where
+    Shm::Builder: Debug,
+{
+    fn managment_segment_name(base_name: &FileName) -> FileName {
+        let origin = "resizable_shared_memory::DynamicMemory::managment_segment_name()";
+        let msg = "Unable to construct management segment name";
+        let mut adjusted_name = *base_name;
+        fatal_panic!(from origin, when adjusted_name.push_bytes(SEGMENT_ID_SEPARATOR),
+                        "This should never happen! {msg} since it would result in an invalid file name.");
+        fatal_panic!(from origin, when adjusted_name.push_bytes(MANAGEMENT_SUFFIX),
+                        "This should never happen! {msg} since it would result in an invalid file name.");
+        adjusted_name
+    }
+
+    fn extract_name_from_management_segment(name: &FileName) -> Option<FileName> {
+        let mut name = *name;
+        if let Ok(true) = name.strip_suffix(MANAGEMENT_SUFFIX) {
+            if let Ok(true) = name.strip_suffix(SEGMENT_ID_SEPARATOR) {
+                return Some(name);
+            }
+        }
+
+        None
```

Comment:
No it is not since the result returns a bool stating `true` or `false` and this indicates if the suffix really existed and was removed or the suffix did not exist and therefore was not removed.